### PR TITLE
fix: remove lib/utils/read-package-name.js

### PR DESCRIPTION
**Pull request by wraithgar**
This code wasn't doing anything special, just dereferencing `name` from
a packument.  There is no need for this to exist.

Most of the tests were able to handle having this go away, except for
`npm owner` which had to have its tests rewritten to be real, which of
course surfaced bugs along the way of behavior that was incorrectly
being tested.  `npm owner` needs some love to clean up its UX, it throws
or returns inconsistently.  I did fix it so that if there is no
package.json in cwd it errored as expected instead of throwing `ENOENT`
which is what it did before.

### CODE REVIEW

**Comment by wraithgar** [comment]
Based off of #4752, that needs to land first

File: lib/commands/diff.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ const pacote = require('pacote')
 const pickManifest = require('npm-pick-manifest')
 const log = require('../utils/log-shim')
 const readPackageName = require('../utils/read-package-name.js')
```
Comment (lukekarrys) [inline lib/commands/diff.js]: It seems like this is used a few other places in this file. Can it be removed from those too?

**Review by lukekarrys — CHANGES_REQUESTED** [review]

File: lib/commands/diff.js (hunk 1 above)
Comment (wraithgar) [reply lib/commands/diff.js]: I thought i got them all. It's very concerning that tests still pass here.

Comment (lukekarrys) [reply lib/commands/diff.js]: also i think the actual util file didn't get deleted, only the test

Comment (wraithgar) [reply lib/commands/diff.js]: This is why we review.  Removed.

Comment (lukekarrys) [reply lib/commands/diff.js]: at least that explains why the tests passed previously 😅

**Review by lukekarrys — APPROVED** [review]

**Pull request merged**